server/src/main/java/org/elasticsearch/action/DocWriteRequest.java

```diff
@@ -30,7 +31,7 @@
  * Generic interface to group ActionRequest, which perform writes to a single document
  * Action requests implementing this can be part of {@link org.elasticsearch.action.bulk.BulkRequest}
  */
-public interface DocWriteRequest<T> extends IndicesRequest, Accountable {
+public interface DocWriteRequest<T> extends IndicesRequest, Accountable, RefCounted {
```

Comment:
These request objects are also used in the high level rest client (and transport client). I think we should add a comment that interacting with the ref counting is not necessary for those clients.

Reply:
++ added a line about this